torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+            missing_str = (
+                f"{', '.join(missing[:-1])} and {missing[-1]}"
+                if len(missing) > 1
+                else missing[0]
+            )
+            raise RuntimeError(
+                f"{missing_str} not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True`"
+                " to automatically download the dataset."
+            )
+
+        # Download missing files
+        record_info = requests.get(self.url).json()["files"]
+        extractable = []
+        for file_info in record_info:
+            file_path = os.path.join(self.root, file_info["key"])
+            if not os.path.exists(file_path):
+                download_url(
+                    file_info["links"]["self"],
+                    self.root,
+                    filename=file_info["key"],
+                    md5=file_info["checksum"].split(":")[1] if self.checksum else None,
+                )
+                if file_info["type"] == "zip":
+                    extractable.append(file_path)
+
+        # Extract downloaded files
+        for file_path in extractable:
+            extract_archive(file_path)
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        show_legend: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            show_legend: flag indicating whether to show a legend for
+                the segmentation masks
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        if "10m_RGB" not in self.bands:
+            raise ValueError("Dataset does not contain RGB bands")
+
+        ncols = self.concat_seasons + 1
+
+        images, mask = sample["image"], sample["mask"]
+        show_predictions = "prediction" in sample
+        if show_predictions:
+            prediction = sample["prediction"]
+            ncols += 1
+
+        start = 0
+        for b in self.bands:
+            if b == "10m_RGB":
+                break
+            start += self.band_nums[b]
+        rgb_indices = [start + s * self.channels for s in range(self.concat_seasons)]
+
+        fig, axs = plt.subplots(nrows=1, ncols=ncols, figsize=(ncols * 4.5, 5))
+        fig.subplots_adjust(wspace=0.05)
+        for ax, index in enumerate(rgb_indices):
+            image = images[index : index + 3].permute(1, 2, 0).numpy()
+            image = percentile_normalization(image)
+            axs[ax].imshow(image)
+            axs[ax].axis("off")
+            if show_titles:
+                axs[ax].set_title(f"Image {ax+1}")
+
+        axs[ax + 1].imshow(mask, vmin=0, vmax=32, cmap=self.cmap)
```

Comment:
We also found that adding `interpolation="none"` to imshow when plotting masks is important, otherwise classes will bleed over when you zoom in.

Reply:
Done